shared/settings/screenprotector.native.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import React, {Component} from 'react'
+import {globalStyles} from '../styles'
+import {Box, Icon, Text, Checkbox} from '../common-adapters'
+import {getSecureFlagSetting, setSecureFlagSetting} from '../native/screenprotector'
+
+type State = {
+  secureFlag: boolean,
+}
+
+class Screenprotector extends Component {
+  state: State = {secureFlag: false}
+  mounted = false
+
+  componentWillMount () {
+    getSecureFlagSetting().then(secureFlag => {
+      this.setState({secureFlag})
+    })
+  }
+
+  componentWillUnmount () {
+    this.mounted = false
+  }
+
+  componentDidMount () {
+    this.mounted = true
+  }
+
+  _changeSecureFlagOption = (nextValue: boolean) => {
+    setSecureFlagSetting(nextValue).then(success => {
+      if (success && this.mounted) {
+        this.setState({secureFlag: nextValue})
+      }
+    })
+  }
+
+  render () {
+    return (
+      <Box style={{...globalStyles.flexBoxColumn, flex: 1, alignItems: 'center', justifyContent: 'center'}}>
```

Comment:
maybe we should sanity check that this is android cause it talks about android. i know you're using the isAndroid flag above but this shouldn't ever render if isAndroid is false